src/dbg/_exports.cpp

```diff
@@ -1653,6 +1653,21 @@ extern "C" DLL_EXPORT duint _dbg_sendmessage(DBGMSG type, void* param1, void* pa
     }
     break;
 
+    case DBG_GET_SYMBOL_INFO_AT:
+    {
+        auto modbase = ModBaseFromAddr((duint)param1);

```

Comment:
This check is already done in `SymbolFromAddressExact`, so it can be removed.

Reply:
But the query is done again, I thought it wouldn't hurt. `ModInfoFromAddr` says it DOES _NOT_ USE LOCKS.

```cpp
MODINFO* ModInfoFromAddr(duint Address)
{
    //
    // NOTE: THIS DOES _NOT_ USE LOCKS
    //
    auto found = modinfo.find(Range(Address, Address));
    ///...
```

Reply:
The extra check doesn’t hurt, but it’s pointless and a (harmless) race condition.

The ModInfoFromAddr can only be used while holding a lock on the modules (hence the comment). This is done with the SHARED_ACQUIRE(?) macro and this ensures the module won’t be unloaded while symbols are being queried.

Reply:
👍 Removed the check.

Reply:
Hm I just saw the issue, but I guess this will work fine.